File: exercises/exercise-1-testing-java/src/main/java/io/miragon/example/adapter/inbound/zeebe/bike/SendBikeWelcomeMailWorker.java
package io.miragon.example.adapter.inbound.zeebe.bike;

import io.miragon.example.adapter.process.generated.BikeSubscriptionSignupProcessApi;
import io.miragon.example.application.port.inbound.bike.SendBikeWelcomeMailUseCase;
import io.miragon.example.domain.bike.BikeSubscriptionId;
import io.camunda.client.annotation.JobWorker;
import io.camunda.client.annotation.Variable;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Component;

import java.util.UUID;

@Component
public class SendBikeWelcomeMailWorker {

    private static final Logger log = LoggerFactory.getLogger(SendBikeWelcomeMailWorker.class);

    private final SendBikeWelcomeMailUseCase useCase;

    public SendBikeWelcomeMailWorker(SendBikeWelcomeMailUseCase useCase) {
        this.useCase = useCase;
    }

    @JobWorker(type = BikeSubscriptionSignupProcessApi.TaskTypes.BIKE_SEND_WELCOME_MAIL)
    public void handle(@Variable String subscriptionId) {
        log.info("Sending welcome mail for subscription: {}", subscriptionId);
        useCase.sendWelcomeMail(new BikeSubscriptionId(UUID.fromString(subscriptionId)));
    }

}

File: exercises/exercise-1-testing-kotlin/src/main/kotlin/io/miragon/example/adapter/inbound/zeebe/bike/SendBikeWelcomeMailWorker.kt
package io.miragon.example.adapter.inbound.zeebe.bike

import io.miragon.example.adapter.process.generated.BikeSubscriptionSignupProcessApi
import io.miragon.example.application.port.inbound.bike.SendBikeWelcomeMailUseCase
import io.miragon.example.domain.bike.BikeSubscriptionId
import io.camunda.client.annotation.JobWorker
import io.camunda.client.annotation.Variable
import mu.KotlinLogging
import org.springframework.stereotype.Component
import java.util.UUID

@Component
class SendBikeWelcomeMailWorker(
    private val useCase: SendBikeWelcomeMailUseCase
) {
    private val log = KotlinLogging.logger {}

    @JobWorker(type = BikeSubscriptionSignupProcessApi.ServiceTasks.BIKE_SEND_WELCOME_MAIL)
    fun handle(@Variable subscriptionId: String) {
        log.info { "Sending welcome mail for subscription: $subscriptionId" }
        useCase.sendWelcomeMail(BikeSubscriptionId(UUID.fromString(subscriptionId)))
    }
}
